### ISSUE TITLE
hybrid k8s federation with azure: cluster offline (dial tcp: i/o timeout)

### ISSUE THREAD

**Issue by clenimar** [Azure/acs-engine#1118]
**Is this a request for help?**: yes

I'm running a Kubernetes federation on-premise, using CoreDNS. The host cluster runs v1.7.1. I've got this container cluster in Azure, v1.6.6, and I'm trying to attach it to the federation. I could join the cluster, but it shows as `Offline` when I list the members of the federation.

I enabled RBAC in the Azure cluster, so I could create clusterroles and bindings.

I can get and create resources in the Azure cluster (using kubectl). I can access the API via curl. But it seems that my federation controller-manager cannot contact Azure cluster's API.

Logs from the Azure kube-apiserver show periodically:
`I0727 07:58:05.787112       1 logs.go:41] http: TLS handshake error from 168.63.129.16:52693: EOF`

Logs from my Federation controller-manager show periodically:
```
E0727 08:33:06.464839       1 cluster_client.go:145] Failed to list nodes while getting zone names: Get https://myk8scluster-myresourcegroup-25ac51.eastus.cloudapp.azure.com/api/v1/nodes: dial tcp: i/o timeout
W0727 08:33:06.464884       1 clustercontroller.go:221] Failed to get zones and region for cluster azure: Get https://myk8scluster-myresourcegroup-25ac51.eastus.cloudapp.azure.com/api/v1/nodes: dial tcp: i/o timeout
```

And the cluster keeps Offline:
```
$ k get clusters --context=fed
NAME               STATUS    AGE
azure              Offline   16h
clenimar-kube-c1   Ready     20h
clenimar-kube-c3   Ready     20h
gke                Ready     20h
```

Any thoughts?
Thank you very much.

**Comment by lachie83** [comment]
I believe there were multiple fixes for RBAC with cluster federation in 1.6.7. (https://github.com/kubernetes/kubernetes/pull/48077). It's probably worth deploying 1.6.7 and determining if your issue still remains

**Comment by clenimar** [comment]
makes sense, I'll give it a go. @lachie83 is there an easy way to upgrade it? or specify the version while deploying a new cluster? searched a bit and it seems that Azure doesn't provide this functionality and I have to do the upgrade manually.
